golem-ts-agentic/packages/golem-ts-sdk/src/internal/clientGeneration.ts

```diff
@@ -139,14 +139,8 @@ class WasmRpxProxyHandlerShared {
     this.agentClassName = agentClassName;
     this.agentType = agentType;
 
-    const constructorParamMeta = AgentConstructorParamRegistry.get(
-      agentClassName.value,
-    );
-    if (!constructorParamMeta) {
-      throw new Error(
-        `No constructor parameter metadata found for ${agentClassName.value}`,
-      );
-    }
+    const constructorParamMeta =
+      AgentConstructorParamRegistry.get(agentClassName.value) ?? new Map();
```

Comment:
Default of new Map() suddenly feels a bit off though. Why was it not in the registry. But may be in the larger context of the code, it's fine.

Reply:
Because if there are no parameters, the registry is never called with this key, so the "ensure" thing is not running either. 
Previously this was not the case because the field was always get directly ony by one, which called the `ensure` and initialized the empty map if needed. But this has been optimized to get the whole map once, here.